### PR TITLE
feat: bump to use node20 runtime

### PR DESCRIPTION
**Pull request by chenrui333**
This is consistent with all the recent github action updates. (mostly in actions repos, like actions/checkout, actions/setup-node, etc.)

-------

Node12 was deleted from runner. Node20 was added to Actions Runner on v2.308.0.
Node16 has en end of life on 11 Sep 2023.

This PR updates the default runtime to node20, rather then node16

relates to https://github.com/actions/runner/pull/2732

### CODE REVIEW

File: .github/workflows/test-action.yml
Comment (chenrui333) [inline .github/workflows/test-action.yml:14]: ubuntu-latest points to 22.04 some while ago, so this is dup now.

File: .github/workflows/test.yml
Comment (chenrui333) [inline .github/workflows/test.yml:29]: leverage the setup-node improvement https://github.com/actions/setup-node/pull/338

**Comment by chenrui333** [comment]
cc @peaceiris (it would also need a major release 3.0.0 after the PR merge)

**Review by peaceiris — DISMISSED** [review]
Thank you so much!

**Comment by peaceiris** [comment]
It will be released as the first version of 2024 together with other bug fixes! ʕ◔ϖ◔ʔ

**Comment by peaceiris** [comment]
👀
https://github.com/peaceiris/actions-hugo/actions/runs/7292054840/job/19882165078?pr=641

**Comment by chenrui333** [comment]
looks like CI is not happy with the dist artifact checkin, I will move it into separate PR.

**Comment by chenrui333** [comment]
@peaceiris works fine now, https://github.com/chenrui333/actions-hugo/actions/runs/7296900196

**Comment by cagataygurturk** [comment]
Any ETA for its release?

**Review by peaceiris — APPROVED** [review]

**Comment by peaceiris** [comment]
Thank you so much! ʕ◔ϖ◔ʔ

**Comment by melhuang** [comment]
do we need to wait for a major version release as well?

**Comment by chenrui333** [comment]
@peaceiris we need a major version release per this merge.

**Comment by peaceiris** [comment]
The tag v3.0.0 was released last week. ʕ◔ϖ◔ʔ

https://github.com/peaceiris/actions-hugo/releases/tag/v3.0.0

**Pull request merged**
